### DISCUSSION TITLE
How to work with JSON columns from JDBC data source

### DISCUSSION THREAD

**Discussion by alex-axiobit** [eclipse-birt/birt#1938] [Q&A]
Where can i find documentation for using Birt Report Designer?

Specifically, right now i am interested on how i could manipulate a JSON object (stored in a column) from a JDBC data source. I need to access it from various elements in the design like Dynamic Text, Chart or Table.

The JSON object can have various properties and can also contain arrays of data that can be represented as tables or charts.

I have found that it's possible to use event handler scripts (Javascript) like beforeOpen, beforeClose, etc. Unfortunately i could not really find any documentation of what you can do inside these handlers and what objects and functions you have access in each handler.

Any help would be much appreciated. Thanks.

**Comment by hvbtup** [comment]
I changed the title to make it more specific. It's a good question.

The `beforeOpen` event of a Data Set can be used to set dataset parameters or to modify the query Text, or for debugging purposes.

The `beforeClose` event is not very useful.

If you select a JSON column (never did this myself), I guess it is comes in as a string. So you could use `eval` to transform it into a JS data structure (Warning: Code Injection possible, but I doubt that BIRT Javascript support special JSON methods for deserialization).

You could use this in the `onFetch` event to set a variable, then use computed columns to extract parts of the data.

For non-scalar data like arrays, I think it gets a bit complicated from here. If you want to use an array as a dataset (e.g. to bind a Table to it), you have to use a Scripted Data Set.

There are examples for using Scripted Data Sets in other discussions (eg #1014).

While BIRT support using XML as as a Data Set, there is no builtin support for using JSON as a Data Set (see #1498 and #985).
